Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -936,6 +936,24 @@ def line_problems(line, _lineno)
 
       return unless @strict
 
+      uses_from_macos_whitelist = %w[
+        perl
+        php
+        python@2
+        ruby
+      ].freeze
+
+      if line =~ /uses_from_macos ["']([^"']+)["']/
```

Comment:
If you can be bothered: ideally this would be a `rubocop` instead of parsing the Ruby with a regex. @GauthamGoli may be able to help if desired. This would mean we can't use `keg_only` stuff but we could just maintain the full whitelist instead (which may be a better idea). Thoughts?

Reply:
@issyl0 If you are trying to check calls to `uses_from_macos` and their args, there's a helper method for writing a RuboCop audit rule using [`find_every_method_call_by_name`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/rubocops/extend/formula.rb#L111), Take a look at some of the existing cops [here](https://github.com/Homebrew/brew/tree/master/Library/Homebrew/rubocops).

Reply:
I had no input into Jon's original implementation of this and I barely understand the regex, so anything simpler sounds good to me! I will :eyes: how to make a `rubocop`.

Reply:
Thanks to help from @GauthamGoli, I've got a basic implementation of a RuboCop cop for this. The test is failing and it only _sort of_ works, but it's a start. I'll come back to it later.